File: node/modelos/civico.js
let lugar = require('./lugar.js');
let datos = require('../datos/civico.json');
class Civico extends lugar{

    constructor(identifier,name,address,review,openingHours){
        super(identifier,name,address,review);
        this.openingHours = openingHours;
        
    }
    getAll(){
        return datos;
    }

    
    getId(id){
        if (id > 0 && id<=datos.length){
            let posicion = id -1;
            let jsonId = datos[posicion];
            return jsonId;
        }
        else{console.log('id incorrecto');}
    }
    
    putId(id,name,address,review,openingHours){
        if (id > 0 && id<=datos.length){
            let posicion = id -1;
            datos[posicion].name = name;
            datos[posicion].address = address;
            datos[posicion].review = review;
            datos[posicion].openingHours = openingHours;
        }
        else{console.log('id incorrecto');}
    }
    
    post(name,address,review,openingHours){
        let id = datos.length + 1;
        let newCiv={'@context': "http://schema.org",'@type': "Accommodation",'identifier': id,'name': name, 'address': address, 'review':review, 'openingHours':openingHours};
        datos.push(newCiv);
    }
    
    deleteId(id){
        if (id > 0 && id<=datos.length){
            let posicion = id -1;
            delete datos[posicion];    
        }
        else{console.log('id incorrecto');}
    }
}

module.exports = Civico;

File: python/modelos/civico.py
from .lugar import Lugar
import json

class Civico:


    def __init__(self,identifier,name,address,review,openingHours):
        Lugar.__init__(self, identifier,name,address,review)
        self.openingHours = openingHours
        with open('datos/civico.json') as data_file:
            self.datos = json.load(data_file)

    def getId(self,iden):
        if (iden > 0 and iden<=len(self.datos)):
            posicion = iden -1
            jsonId = self.datos[posicion]
            return jsonId
        
        else:
            return 'id incorrecto'

    def deleteId(self,iden):
        if (iden > 0 and iden<=len(self.datos)):
            posicion = iden -1
            self.datos[posicion] = "Eliminado"
            return self.datos
        
        else:
            return 'id incorrecto'

    def post(self,name,address,review,openingHours):
        posicion = len(self.datos) 
        iden = posicion + 1
        entrada=["@context: http://schema.org","@type: Accommodation","identifier:" + str(iden), "name:" + name, "address:" + address, "review:" + review, "openingHours:" + openingHours ]
        self.datos.append(entrada)
        return self.datos

    def putId(self,iden,name,address,review,openingHours):
        if (iden > 0 and iden<=len(self.datos)):
            posicion = iden -1
            self.datos[posicion]["name"] = name
            self.datos[posicion]["address"] = address
            self.datos[posicion]["review"] = review
            self.datos[posicion]["openingHours"] = openingHours
            return self.datos[posicion]
        
        else:
            return 'id incorrecto'       
    